### PR TITLE
ci, docs: add linkcheck job, fix links

### LINKED ISSUE #1704: some external links are broken
## 📚 Documentation

Sphinx has a builder for checking external links. You can run with `make linkcheck` in the docs directory. And, there are some broken links.
```
about.rst:22: [broken] governance.html: 
contrib/handlers.rst:93: [broken] https://docs.neptune.ai/python-api/tutorials/get-started.html#copy-api-token: Anchor 'copy-api-token' not found
contrib/handlers.rst:72: [broken] https://docs.polyaxon.com/references/polyaxon-tracking-api/experiments/: 404 Client Error: Not Found for url: https://polyaxon.com/docs/references/polyaxon-tracking-api/experiments/
contrib/handlers.rst:72: [broken] https://docs.polyaxon.com/references/polyaxon-tracking-api/experiments/: 404 Client Error: Not Found for url: https://polyaxon.com/docs/references/polyaxon-tracking-api/experiments/
contrib/handlers.rst:76: [broken] https://github.com/facebookresearch/visdom#visdom-arguments-python-only: Anchor 'visdom-arguments-python-only' not found
contrib/handlers.rst:139: [broken] https://docs.polyaxon.com/references/polyaxon-tracking-api/experiments/: 404 Client Error: Not Found for url: https://polyaxon.com/docs/references/polyaxon-tracking-api/experiments/
distributed.rst:34: [broken] https://horovod.readthedocs.io/en/latest/api.html#horovod.torch.init: Anchor 'horovod.torch.init' not found
engine.rst:113: [broken] https://github.com/pytorch/ignite/tree/master/examples/contrib/cifar10#check-resume-training: Anchor 'check-resume-training' not found
engine.rst:242: [broken] https://github.com/pytorch/ignite/tree/master/examples/contrib/cifar10#check-resume-training: Anchor 'check-resume-training' not found
engine.rst:112: [broken] https://github.com/pytorch/ignite/tree/master/examples/mnist#training-save--resume: Anchor 'training-save--resume' not found
engine.rst:241: [broken] https://github.com/pytorch/ignite/tree/master/examples/mnist#training-save--resume: Anchor 'training-save--resume' not found
faq.rst:85: [broken] _modules/ignite/contrib/engines/tbptt.html#create_supervised_tbptt_trainer: 
```
To have an updated link to check, we could do
- add a github action job in `docs.yml` checking `make linkcheck` running on every PR
- update the links with new links

### PR DESCRIPTION
**Pull request by ydcjeff**
Fixes #1704 

Description:
- add `linkcheck` job
- change broken links
- GitHub handles the anchor links by loading some JS, so sphinx fails to detect it. Answer is append `user-content-` in front of anchors. https://github.com/sphinx-doc/sphinx/issues/6779#issuecomment-548171598
- horovod broken link https://github.com/horovod/horovod/issues/2695

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [x] Documentation is updated (if required)

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -34,3 +36,23 @@ jobs:
...
+        with:
+          path: ~/.cache/pip
+          key: pip-${{ hashFiles('requirements-dev.txt') }}-${{ hashFiles('docs/requirements.txt') }}
+
+      - name: Install docs deps
+        run: bash .github/workflows/install_docs_deps.sh
+
+      - name: make linkcheck
+        working-directory: ./docs/
+        run: make linkcheck --jobs 2 SPHINXOPTS="--color"
```
Comment (ydcjeff) [inline .github/workflows/docs.yml:58]:
```suggestion
        run: make linkcheck --jobs 2 SPHINXOPTS="--color -W"
```

**Comment by ydcjeff** [comment]
@vfdev-5 
ready to review!

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -3,11 +3,13 @@ name: Build docs
 on:
   push:
     branches: [master]
+  pull_request:
   release:
     types: [published]
 
 jobs:
   build-deploy:
+    if: github.event_name == 'push'
```
Comment (vfdev-5) [inline .github/workflows/docs.yml:12]: Shouldn't this also include release event ?

**Review by vfdev-5 — COMMENTED** [review]
Thanks for the update @ydcjeff ! 
I left a comment.

File: .github/workflows/docs.yml (hunk 2 above)
Comment (ydcjeff) [reply .github/workflows/docs.yml]: Yes, I missed this

**Review by vfdev-5 — APPROVED** [review]
Thanks a lot for the PR @ydcjeff !
Hopefully, we should now have our link correct

**Pull request merged**